client-js/app/scripts/obligation.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Obligation is the input side complement of Promise; if one party is promised
+ * a value, then of necessity another party must be obliged to provide the
+ * value or explain why they failed to do so. Obligation has two methods
+ * `resolve` and `reject` which can be used to send a value or an error into a
+ * chain of asynchronous callbacks. Promise is the output side of the chain,
+ * providing access to the asynchronous values by adding callback functions.
+ *
+ * In the traditional ES6 Promise API, resolve and reject functions are only
+ * available as arguments passed in to the executor function we pass to the
+ * Promise constructor. But in many situations we need to store the callbacks
+ * for later use or access them outside the constructor, which Obligation makes
+ * possible. This is especially useful in situations where we need to interface
```

Comment:
Also, you say `Promise` has no `resolve` and `reject` methods, but then [Mozilla says otherwise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise). Is there a reference that's not the completely unreadable official ES6 spec?